scripts/lis_hyperv_platform/main.ps1

```diff
@@ -197,19 +199,14 @@ function Main {
         if ($LASTEXITCODE) {
             throw "Ssh connection failed with error code: $LASTEXITCODE"
         }
-        if ($kernelRevision -like "*$kernelTag*") {
+        if (($kernelRevision -like "*$kernelTag*") -or ($kernelRevision -like "*next*")) {
             Write-Host "Kernel $kernelRevision matched"
         } else {
             throw "Could not find the kernel: $kernelTag"
         }
     }
     $JobManager.RemoveTopic($InstanceName)
 
-    Write-Host "Copying boot logs to workspace..."
```

Comment:
do not remove this

Reply:
the boot log needs to be copied also if the kernel has not been found, this was the bug. you can do a try/catch finally for the kernel check and wait ip for vm and if something fails or goes well, the boot logs will be copied.